dakota_manylinux_install_files/CMakeLists.txt

```diff
@@ -0,0 +1,318 @@
+# Sets several convenience variables
+# NOTE: depends on presence of generated source directories
+if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/VERSION)
+  if(NOT EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/src/DakotaBuildInfo.cpp)
+    message(FATAL_ERROR "VERSION file exists but no corresponding
+                         src/DakotaBuildInfo.cpp exists.")
+  endif()
+  set(DAKOTA_VERSION_FOUND TRUE)
+else()
+  dakota_version_from_git()
+  dakota_create_version_file()
+endif()
+
+
+include(DakotaCPack)
+
+# The permission level of the distribution (to be made an option)
+# TODO: change default to DAKOTA_ExternalWeb?
+# BMA TODO: Publish this option in options directory
+if(NOT DEFINED DAKOTA_DISTRO)
+  set(DAKOTA_DISTRO "snlfull" CACHE STRING "Dakota Distribution Type")
+endif()
+
+# Initial CPack settings and source exclusions
+# Depends on DAKOTA_DISTRO for file filtering
+dakota_cpack_initialize()
+
+# Prune local and non-distributable TPLs based on DAKOTA_DISTRO
+dakota_cpack_prune_distro()
+
+# Default RPATH settings
+#
+# These settings must take into consideration the presence of shared
+# libraries in both bin/ and /lib, and applications in bin/ (formerly
+# test/, which is now managed in share/dakota/test; see override in
+# test/CMakeLists.txt).
+#
+#
+# $ORIGIN (@executable_path on Mac) is replaced at runtime by the path
+# of the executable (application or .so).
+#
+# Add:
+#   $ORIGIN so executables can find shared libs in their own folder
+#   $ORIGIN/../lib so that executables in bin and test can find shared libs in the lib folder
+#   $ORIGIN/../bin so that exectuables in lib and test can find shared libs in the bin folder.
+if(APPLE)
+  set(CMAKE_MACOSX_RPATH TRUE CACHE BOOL "Add @rpath to library install names")
+  set(CMAKE_INSTALL_RPATH "@executable_path;@executable_path/../lib;@executable_path/../bin"
+    CACHE STRING "Set the RPATH in Dakota executables and libraries")
+elseif(UNIX)
+   set(CMAKE_INSTALL_RPATH "\$ORIGIN:\$ORIGIN/../lib:\$ORIGIN/../bin"
+     CACHE STRING "Set the RPATH in Dakota executables and libraries")
+endif()
+
+
+# Now check languages after compiler flags and options are set
+enable_language(C)
+enable_language(CXX)
+enable_language(Fortran)
+if(MSVC_VERSION EQUAL 1400 OR MSVC_VERSION GREATER 1400 OR MSVC10)
+  add_definitions(-D_CRT_SECURE_NO_DEPRECATE -D_CRT_NONSTDC_NO_DEPRECATE 
+    -D_CRT_SECURE_NO_WARNINGS)
+  add_definitions(-D_SCL_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_WARNINGS)
+endif()
+
+# Perl is required for examples, docs, and system tests
+find_package(Perl REQUIRED)
+
+set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wall -lpthread -lutil")
+# Python is optionally required by Dakota, Teuchos, and Acro; probe
+# for components here at top-level:
+include_directories(${PYTHON_INCLUDE_DIRS})
+include_directories(/usr/include)
+include_directories(/usr/lib64)
+include_directories(/tmp/INSTALL_DIR/dakota-6.18.0-public-src-cli/packages/external/eigen3/include/)
+find_package(Threads REQUIRED)
```

Comment:
Checked against `stable_2023_10_24` and the file only had a minor addition to cover a small bug.
This is nice regardless of patching or file replacement I guess.

```diff
--- dakota-6.18.0-public-src-cli/CMakeLists.txt	2023-05-11 03:08:25
+++ dakota-stable_2023_10_24-public-src-cli/CMakeLists.txt	2023-10-24 08:11:21
@@ -186,6 +186,13 @@
       "${CMAKE_SHARED_LIBRARY_CREATE_CXX_FLAGS} -undefined dynamic_lookup")
 endif()

+# set compiler flag to prevent issues in boost::serialization for c++17.
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+  if(CMAKE_CXX_STANDARD GREATER 11)
+    set(CMAKE_CXX_FLAGS "-fno-new-ttp-matching ${CMAKE_CXX_FLAGS}"
+        CACHE STRING "fix boost::serialization" FORCE)
+  endif()
+endif()
```